### PR TITLE
chore: unify rendering of Recover Banner

### PR DESCRIPTION
**Pull request by gre**
### 📝 Description

This makes sure that only one banner at a time is displayed, before this fix, the recover banner could appear on top of the content card banner, instead we have made it so the recover banner is first displayed before the second. This is implemented with a "passthrough" pattern (the RecoverBanner will render any children that is passed if it doesn't need to render anything, that way RecoverBanner remains decoupled from ContentCards)

On top of this, we have fixed inconsistencies in the UI of the Recover Banner that was fixed on this PR by making RecoverBanner renders an ActionCard. a leftContent generic prop was introduced to allow to inject custom left content instead of an image.

⚠️  I have only tested by modifying directly into the code to force enable recover banner because I couldn't properly access the dev environment of recover. Another QA / dev will need to take over to confirm the banner still works as attended, and notably in combination with Braze banners.

```
    //const storage = await getStoreValue("SUBSCRIPTION_STATE", protectID);
    //const displayBanner = await getStoreValue("DISPLAY_BANNER", protectID);
    const storage = "STARGATE_SUBSCRIBE";
    const displayBanner = "true";
```

https://github.com/LedgerHQ/ledger-live/assets/211411/12ed9ec6-b2f6-424b-bf09-d118dddb9953

### ❓ Context

- **JIRA or GitHub link**: [LIVE-11239]

### ✅ Checklist

Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.

- [x] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** we are far from being able to automate test for this unfortunately. but we should plan this – cc @KVNLS  
- [x] **Impact of the changes:** 
  - the Recover Banner and its behaviour when there is also content cards banner

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.
- [ ] **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-11239]: https://ledgerhq.atlassian.net/browse/LIVE-11239?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Justkant — APPROVED** [review]

**Comment by cgrellard-ledger** [comment]
Is it expected that we don't have the purple background in the recover card now ?

**Comment by gre** [comment]
Capture d’écran 2024-03-12 à 17 05 42

fixed

**Comment by cgrellard-ledger** [comment]
Can we make sure with @Anthony-Ledger that we want to hide the action cards and the portfolio carousel when the recover banner is displayed ?

**Comment by gre** [comment]
@cgrellard-ledger it's what the spec says. "Recover takes the lead (depends what’s easier technically)"

**Comment by cgrellard-ledger** [comment]
> @cgrellard-ledger it's what the spec says. "Recover takes the lead (depends what’s easier technically)" i think we can discuss this indeed (NB: my next working day is Monday)

Yep let's discuss this, I'm not sure how to understand the "takes the lead". It will an impact on the growth team if we do it this way. I'll get Anthony's input just to confirm

**Comment by cgrellard-ledger** [comment]
Anthony is fine with this solution ✅

**Review by cgrellard-ledger — DISMISSED** [review]

**Review by Justkant — APPROVED** [review]

**Review by cgrellard-ledger — DISMISSED** [review]

File: apps/ledger-live-desktop/src/renderer/components/RecoverBanner/RecoverBanner.tsx [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
+const Wrapper = styled(Card)`
   background-color: ${p => p.theme.colors.opacityPurple.c10};
+  margin: 20px 0px;
 `;
 
-export default function RecoverBanner() {
+/**
+ * @prop children: if a child is passed, it will be rendered instead of the default banner. this allows to do a passthrough to have first the recover banner, then the rest of the content.
+ */
+export default function RecoverBanner({ children }: { children?: React.ReactNode }) {
```
Comment (Justkant) [inline apps/ledger-live-desktop/src/renderer/components/RecoverBanner/RecoverBanner.tsx:21]: You can use this type too if you want

```suggestion
export default function RecoverBanner({ children }: React.PropsWithChildren) {
```

**Review by Justkant — APPROVED** [review]
LGTM just a small suggestion if you want to use it

**Review by cgrellard-ledger — APPROVED** [review]

**Pull request merged**